studentPerformanceAnalyzer.c

Keerthi2181 opened the pull request.

Keerthi2181 committed:
Create studentPerformanceAnalyzer.c

intimetec-shiv-maheshwari commented on studentPerformanceAnalyzer.c:
```diff
@@ -0,0 +1,90 @@
+    int n;
+    
+    printf("Enter number of Students:");
+    scanf("%d",&n);
```
What is n ? Please use proper naming conventions. I have already made it clear in the first assignment.

intimetec-shiv-maheshwari commented on studentPerformanceAnalyzer.c:
```diff
@@ -0,0 +1,90 @@
+    scanf("%d",&n);
+    struct Student s[n];
+    for(int i=0;i<n;i++){
+        printf("Enter details of student%d:\n",i+1);
```
There is no where handling of corner cases. You have just done the assignment for the sake of doing it. I will not review it until you handle the corner cases properly. Re-write the whole logic with proper edge case handling.
Eg. Why i am able to enter roll number as -ve integer. Think like this and you will be able to figure out the edge cases automatically.

Keerthi2181 committed:
Update studentPerformanceAnalyzer.c

Input validations included
1.Roll number validation: must be positive.
2.Marks validation: can only be between 0 and 100.
3.Name validation: can contain only alphabets and spaces.
4.Number of students: must be positive.

Keerthi2181 committed:
Update studentPerformanceAnalyzer.c

Roll number- alphabets not allowed
Number of students- only positive numbers allowed
Duplicate roll numbers handled
Used camel casing
Added separate functions for each part
Made main function simple
